### PR TITLE
CRDCDH-921 Creating Data View Details dialog

### PR DESCRIPTION
**Pull request by Alejandro-Vega**
### Overview

Created a dialog that will display more details about nodes in the Data View tab of the Data Submission Dashboard. When the identity column (first column) is clicked, it will open this dialog to show more information as well as the related nodes displayed in table form.

### Change Details (Specifics)

- Added queries/types for `getNodeDetail` and `getRelatedNodes`. Also created `getRelatedNodeProperties` query to get the properties and identity column separately. 
- Updated getSubmissionNodes query to return the `IDPropName`, which I used to always ensure the identity column is first in the data view table.
- Created data view details dialog which consists of information about the node itself, as well as related nodes displayed as tabs. It will show the parent nodes and child nodes as tabs, with the parent nodes being first, as well as the total nodes being displayed.
- Didn't allow the dialog to be displayed for "data file" node type, this is described in another ticket but just added it in since it caused errors otherwise. ([CRDCDH-1359](https://tracker.nci.nih.gov/browse/CRDCDH-1359))
- Added custom scrollbar for the table, per design, although we should consider standardizing the scrolling for all tables.
- Added tests for the dialog, data view details section, as well as the related nodes (tabs/table) portion.
- Fixed data view table not showing the correct default sorting/sort direction. Although, the data is being returned twice, might need to update the query to first return columns, then render the table.

> [!NOTE]
> If you click on a tab in the dialog and it has no content, this is probably because it had issues finding the related nodes, which are described in the validation results and is expected behavior.

### Related Ticket(s)

[CRDCDH-921](https://tracker.nci.nih.gov/browse/CRDCDH-921)

### CODE REVIEW

**Review by amattu2 — CHANGES_REQUESTED** [review]
Awesome job. I annotated a few things below, most of which are optional updates, so feel free to dismiss them.

File: src/components/DataSubmissions/DataViewDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    <Stack direction="column">
+      <Stack direction="row" alignItems="flex-start">
+        <StyledNodeTypeWrapper direction="column">
+          <StyledNodeTypeLabel variant="h5">Node Type</StyledNodeTypeLabel>
+          <StyledNodeTypeValue variant="h6">{nodeType}</StyledNodeTypeValue>
+        </StyledNodeTypeWrapper>
+        <StyledSectionDivider orientation="vertical" />
+        <Grid container flexDirection="row" rowSpacing="3px">
+          <Grid xs={2} item>
+            <StyledLabel variant="body1">{nodeType} ID</StyledLabel>
```
[the remark this answers is not in the document]
Comment (amattu2) [reply src/components/DataSubmissions/DataViewDetails.tsx:126]: Retested and the overlap is fixed.

Screenshot 2024-07-17 at 3 32 38 PM

File: src/graphql/getRelatedNodeProperties.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import gql from "graphql-tag";
+
+export const query = gql`
+  query getRelatedNodeProperties(
```
[the remark this answers is not in the document]
Comment (Alejandro-Vega) [reply src/graphql/getRelatedNodeProperties.ts]: Thanks for this suggestion, I added it!

File: src/components/DataSubmissions/DataViewDetailsDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+const DataViewDetailsDialog = ({
+  submissionID,
+  nodeType,
+  nodeID,
+  open,
+  closeText = "Close",
+  onClose,
+  ...rest
+}: Props) => (
+  <StyledDialog open={open} onClose={onClose} title="" scroll="body" {...rest}>
```
[the remark this answers is not in the document]
Comment (Alejandro-Vega) [reply src/components/DataSubmissions/DataViewDetailsDialog.tsx:64]: Updated it with some of your help, and made my own adjustments. Had to make some changes due to certain parts breaking. The initial load is still not perfect, but switching tabs is definitely smoother.

**Review by amattu2 — APPROVED** [review]
**Substantially** smoother now. Awesome job!

**Comment by amattu2** [comment]
Deleting branches:
- https://github.com/CBIIT/crdc-datahub-ui/tree/CRDCDH-921-temp
- https://github.com/CBIIT/crdc-datahub-ui/tree/amattu2/CRDCDH-921-Fixes

**Pull request merged**